src/mongocxx/options/find.hpp

```diff
@@ -113,6 +113,21 @@ class MONGOCXX_API find {
     const stdx::optional<enum cursor_type>& cursor_type() const;
 
     ///
+    /// Sets the index to use for this operation.
+    ///
+    /// @param index_hint
+    ///   Object representing the index to use.
+    ///
```

Comment:
Add `@see https://docs.mongodb.org/manual/reference/operator/meta/hint/`.
Should we add "When an index filter exists for the query shape, MongoDB ignores the $hint." as a note?